torch/csrc/jit/passes/symbolic_shape_analysis.cpp

```diff
@@ -43,6 +43,91 @@ static bool symbolic_shape_analysis_test_mode = false;
 namespace torch {
 namespace jit {
 
+// This is similar to c10::SymbolicShape, but instead of either having
+// a concrete dimension or a symbolic dimension, an argument may be:
+// - A Symbolic Dimension
+// - A Constant Integer
+// - Neither of the above. The third case can occur due to inputs to
+// ops like view that accept negative values. Maintaining the distinction
+// between an unknown symbolic dimension and an unknown integer allows
```

Comment:
unkown symbolic dimension is just a fresh symbolic dimension?

Reply:
yes, that's correct. maybe `unknown dimension`